src/event-bridge/event-bus.ts

```diff
@@ -83,7 +113,109 @@ abstract class EventBusBase<E extends EventBusRuleInput>
   public static readonly FunctionlessType = "EventBus";
   readonly functionlessKind = "EventBus";
 
-  constructor(readonly bus: aws_events.IEventBus) {}
+  readonly bus: aws_events.IEventBus;
+
+  constructor(bus: aws_events.IEventBus) {
+    this.bus = bus;
+    return makeCallable(this, (call: CallExpr, context: CallContext) => {
+      if (isASL(context)) {
+        this.bus.grantPutEventsTo(context.role);
+
+        // Lets validate and normalize that the events are
+        const eventObjs = call.args.reduce(
+          (events: ObjectLiteralExpr[], arg) => {
+            if (isArrayLiteralExpr(arg.expr)) {
+              if (
+                !arg.expr.items.every((item): item is ObjectLiteralExpr =>
+                  isObjectLiteralExpr(item)
+                )
```

Comment:
```
.every(isObjectLiteralExpr)
```

Reply:
Thanks. Prefer this form too, not sure why I missed it...